app/src/main/java/com/rockthevote/grommet/ui/registration/RegistrationDataTransformer.kt

```diff
@@ -0,0 +1,353 @@
+        val phoneCapabilities = listOfNotNull(
+            "voice",
+            if (additionalInfoData.phoneType == PhoneType.MOBILE) "sms" else null
+        )
+
+        val phone = ContactMethod(
+            type = "phone",
+            value = additionalInfoData.phoneNumber,
+            capabilities = phoneCapabilities
+        )
+
+        val email = ContactMethod(
+            type = "email",
+            value = additionalInfoData.emailAddress,
+            capabilities = listOf()
+        )
+
+        return listOf(phone, email)
+    }
+
+    private fun buildAdditionalInfo(): List<AdditionalInfo> {
+        val preferredLanguage = AdditionalInfo(
+            name = "preferred_language",
+            stringValue = additionalInfoData.preferredLanguage?.toString()
+        )
+
+        // TODO confirming, is assistant_declaration based on "I CONFIRM THAT I HAVE... in assistant, or just "Did someone help you with this form"?
+        val assistantDeclaration = AdditionalInfo(
+            name = "assistant_declaration",
+            stringValue = if (assistanceData.hasSomeoneAssisted) {
+                if (assistanceData.hasConfirmedTerms) {
+                    "true"
+                } else {
+                    throw InvalidRegistrationException(
+                        "assistanceData.hasSomeoneAssisted is true but, did not confirm terms",
+                        R.string.confirm_terms_not_checked
+                    )
+                }
+            } else {
+                "false"
+            }
+        )
+
+        return listOf(preferredLanguage, assistantDeclaration)
+    }
+
+    private fun PersonNameData.toApiName(): Name {
+        return Name(
+            firstName = firstName,
+            lastName = lastName,
+            middleName = middleName,
+            titlePrefix = title.toString(), // TODO should this be localized?
+            titleSuffix = suffix?.toString() // TODO should this be localized?
+        )
+    }
+
+    private fun AddressData.toApiAddressData(): Address {
+        val base = NumberedThoroughfareAddress(
+            completeAddressNumber = "",
+            completeStreetName = streetAddress,
+            completeSubAddress = buildSubAddress(),
+            completePlaceNames = buildCompletePlacesNames(),
+            state = state,
+            zipCode = zipCode
+        )
+
+        return Address(base)
+    }
+
+    private fun AddressData.buildSubAddress(): List<CompleteSubAddress>? {
+        val unitType = unitType?.let {
+            if (it.isEmpty()) return@let null
+
+            val subAddressComponents = listOfNotNull(unitType, unitNumber)
+                .nullIfEmpty() ?: return@let null
+
+            CompleteSubAddress(
+                subAddressType = it,
+                // TODO Do we use the unitNumber as is, or append a transformed unitType like in the JSON example?
```

Comment:
unit number as is, API spec says "include full text the user inputs (even if it includes "apt")"

So , we don't append the unit type, but if they do something like select "APT" as the unit type and then also type "APT 123", then we would be sending **APT** as the type and "APT 123" as the number.